crates/noirc_frontend/src/hir/mod.rs

```diff
@@ -76,28 +102,28 @@ impl Context {
     pub fn get_all_test_functions_in_crate_matching(
         &self,
         crate_id: &CrateId,
-        pattern: &str,
+        pattern: FunctionNameMatch,
     ) -> Vec<(String, FuncId)> {
         let interner = &self.def_interner;
         let def_map = self.def_map(crate_id).expect("The local crate should be analyzed already");
 
         def_map
             .get_all_test_functions(interner)
             .filter_map(|id| {
-                let name = interner.function_name(&id);
-
-                let meta = interner.function_meta(&id);
-                let module = self.module(meta.module_id);
-
-                let parent = def_map.get_module_path_with_separator(
-                    meta.module_id.local_id.0,
-                    module.parent,
-                    "::",
-                );
-                let path =
-                    if parent.is_empty() { name.into() } else { format!("{parent}::{name}") };
-
-                path.contains(pattern).then_some((path, id))
+                let fully_qualified_name = self.fully_qualified_function_name(crate_id, &id);
+                match &pattern {
+                    FunctionNameMatch::Anything => Some((fully_qualified_name, id)),
+                    FunctionNameMatch::Exact(pattern) => {
+                        if &fully_qualified_name == pattern {
+                            Some((fully_qualified_name, id))
+                        } else {
+                            None
+                        }
```

Comment:
```suggestion
                        (&fully_qualified_name == pattern).then_some((fully_qualified_name, id))
```

Reply:
TIL! I figured that `contains` returned an option and this was an available function, didn't realize it was implemented on bools 🎉